File: homework/src/main/java/com/mikheev/homework/domain/Genre.java
package com.mikheev.homework.domain;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "id")
@Document(collection = "genres")
public class Genre {

    @Id
    private String id;

    private String name;
}

File: homework/src/main/kotlin/com/mikheev/homework/domain/Genre.kt
package com.mikheev.homework.domain

import jakarta.persistence.*

@Entity
@Table(name = "genres")
class Genre {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @Column(name = "name")
    val name: String? = null
}
